packages/react/src/transactions/adapters/__tests__/CollectPublicationCallGateway.spec.ts

```diff
@@ -81,4 +133,107 @@ describe(`Given an instance of the ${CollectPublicationCallGateway.name}`, () =>
       expect(unsignedCall.nonce).toEqual(nonce);
     });
   });
+
+  describe(`when calling the "${CollectPublicationCallGateway.prototype.createProxyTransaction.name}" method`, () => {
+    describe('when relaying a FreeCollectRequest', () => {
+      describe('and receiving a successful response', () => {
+        const request = mockFreeCollectRequest();
+        it(`should resolve with ${ProxyTransaction.name} on Polygon`, async () => {
+          const indexingId = 'indexing-id';
+          const apollo = createMockApolloClientWithMultipleResponses([
+            createBroadcastProxyActionCallMockedResponse({
+              result: indexingId,
+              variables: {
+                request: {
+                  collect: {
+                    freeCollect: {
+                      publicationId: request.publicationId,
+                    },
+                  },
+                },
+              },
+            }),
+          ]);
+
+          const mockTransactionObserver = mockITransactionObserver();
+          const factory = mockITransactionFactory(mockTransactionObserver);
+          const collectPublicationCallGateway = mockCollectPublicationCallGateway({
+            apollo,
+            factory,
+          });
+
+          const transactionResult = await collectPublicationCallGateway.createProxyTransaction(
+            request,
+          );
+
+          if (transactionResult.isFailure()) throw transactionResult.error;
+
+          const transaction = transactionResult.value;
+
+          await transaction.waitNextEvent();
+
+          expect(transaction).toEqual(
+            expect.objectContaining({
+              chainType: ChainType.POLYGON,
+              id: 'id',
+              request,
+              status: ProxyActionStatus.MINTING,
+            }),
+          );
+
+          // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-explicit-any
+          expect((transaction as any).state.proxyId).toEqual(indexingId);
+        });
+      });
+      describe('and receiving an error response', () => {
+        const request = mockFreeCollectRequest();
+        it(`should resolve with ${ProxyTransaction.name} on Polygon`, async () => {
```

Comment:
This wording here does not reflect the actual expectations in the test code.

Accounting for the previous comments let's try to rewrite these so that we have less nesting (seems not needed in this case) and the whole test reads like a specification for this unit:

```
when calling the "createProxyTransaction" method
  [it] should succeed with a "ProxyTransaction" on Polygon
  [it] should fail with a "BroadcastError" in case of broadcast failure
```

another way would be to shift the "when" into a precondition (part of the "givens") and nest one more time:
```
and a "FreeCollectRequest" via the "createProxyTransaction" method
  when the broadcasting succeeds
    [it] should succeed with a "ProxyTransaction" on Polygon
  when the broadcasting fails
    [it] should fail with a "BroadcastError" in case of broadcast failure
```

"succeed" and "fail" aims at suggesting this it returning `Success<ProxyTransaction>` and and `Failure<BroadcastError>` (I took some shortcuts in writing these types to get to the point). We use these in other places. The intent it's to make a distinction between succeeds/fails vs. resolves/rejects vs. returns/throws. Not sure if very intelligible, but we use this elsewhere.

Reply:
(please don't mind by broken english... take my wording as an example but improve it if needs to)

Reply:
okay nice, thanks for detailed response here, amended 👍